lib/httparty/request.rb

```diff
@@ -31,20 +31,30 @@ class Request #:nodoc:
 
     def initialize(http_method, path, o = {})
       self.http_method = http_method
-      self.path = path
       self.options = {
         limit: o.delete(:no_follow) ? 1 : 5,
         assume_utf16_is_big_endian: true,
         default_params: {},
         follow_redirects: true,
         parser: Parser,
+        uri_adapter: URI,
         connection_adapter: ConnectionAdapter
       }.merge(o)
+      self.path = path
       set_basic_auth_from_uri
     end
 
     def path=(uri)
-      @path = URI(uri)
+      uri_adapter = options[:uri_adapter]
+
+      @path = if uri.is_a?(uri_adapter)
+        uri
+      elsif String.try_convert(uri)
```

Comment:
But that's not exactly what it's doing right now. I mean might break backwards compatibility, right?